### PR TITLE
feature/authentication

### PR DESCRIPTION
**Pull request by Kevin-Solomon**
Features Added :
- [x] Add Login feature.
- [x] Add Signup feature.
- [x] Made a custom ```useForm``` hook for form submission.

Learnings from creating this feature.
- Invalid hook calls.

## Right now if the response is 200 it redirects to the landing page.

### CODE REVIEW

File: src/pages/Login/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { useForm } from '../../hooks';
+import { Link, Navigate } from 'react-router-dom';
+import { useAuth } from '../../context/auth/authContext';
+import './Login.css';
+import { useNavigate } from 'react-router-dom';
```
Comment (Sudhanshusoni45) [inline src/pages/Login/Login.jsx]: you can shift this import to line number 3 along with Link

File: src/pages/Login/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { useForm } from '../../hooks';
+import { Link, Navigate } from 'react-router-dom';
```
Comment (Sudhanshusoni45) [inline src/pages/Login/Login.jsx]: How are you importing Navigate from react router dom, we import useNavigate right?

File: src/pages/Signup/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { useForm } from '../../hooks';
+import { Link } from 'react-router-dom';
+import { formSubmitHandler } from '../../util';
+import { useAuth } from '../../context/auth/authContext';
+import { useNavigate } from 'react-router-dom';
```
Comment (Sudhanshusoni45) [inline src/pages/Signup/Signup.jsx]: Here also you can move this import to line 3

**Review by Sudhanshusoni45 — APPROVED** [review]
Suggested just some refactoring.
Code is well written neat and clean.

File: src/hooks/useForm.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { useState } from 'react';
+
+const useForm = initialState => {
+  const [userInput, setUserInput] = useState(initialState);
+  const changeHandler = e => {
```
Comment (starver20) [inline src/hooks/useForm.js:5]: Nice custom hook, you can also try adding validation to the same hook.

File: src/pages/Login/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+export const Login = () => {
+  const [userInput, setUserInput] = useForm({ email: '', password: '' });
+  const { setUser } = useAuth();
+  const navigate = useNavigate();
+  return (
+    <div className="login-container">
+      <div className="login-card">
+        <h2>Login</h2>
+        <form
+          onSubmit={async e => {
```
Comment (starver20) [inline src/pages/Login/Login.jsx:17]: You can extract this into a seperate function.

File: src/pages/Signup/Signup.jsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    password: '',
+  });
+  const { user, setUser } = useAuth();
+  console.log(user);
+  return (
+    <div className="login-container">
+      <div className="login-card">
+        <h2>Sign Up</h2>
+        <form
+          onSubmit={async e => {
```
Comment (starver20) [inline src/pages/Signup/Signup.jsx:20]: Similarly this can also be extracted into a function

**Review by starver20 — APPROVED** [review]
Gave a few suggestions. Other than that Code looks good to merge.

File: src/pages/Signup/Signup.jsx (hunk 6 above)
Comment (Kevin-Solomon) [reply src/pages/Signup/Signup.jsx:20]: If I extract the function I will get an invalid hook call error.
As hooks need to be called in a functional component.

**Pull request merged**